src/lib/Semantics.ml

```diff
@@ -1020,6 +1111,27 @@ and do_rigid_cap r s phi code =
         throw @@ NbeFailed "do_rigid_cap"
     end
 
+and do_rigid_vproj r v =
+  let open CM in
+  abort_if_inconsistent D.Abort @@
+  begin
+    inspect_con v |>>
+    function
+    | D.Cut {tp = D.El vtp; cut} ->
+      inspect_con vtp |>>
+      begin
+        function
+        | D.CodeV (_,pcode,code,pequiv) ->
```

Comment:
Why do we need to check the type? This doesn't seem analogous to what we're doing in `do_rigid_cap`; can we pass the necessary data to `do_rigid_vproj`?

Reply:
for neutrals, we always obtain the necessary information from the type. furthermore, we will not always have access to the relevant information when we call this function, so it is better to do it this way.

Reply:
Compare `fst`, `snd`.